src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  return sourceString
+    .split(';')
+    .map((keyVal) => keyVal.split(':').map((str) => str.trim()))
+    .filter(([key, value]) => key && value)
```

Comment:
The use of `map` and `filter` here is appropriate as they do not mutate the original array. The chaining of these methods is also correctly indented, adhering to good code style practices.